exec/java-exec/src/main/java/org/apache/drill/exec/expr/stat/RangeExprEvaluator.java

```diff
@@ -69,70 +69,70 @@ public long getRowCount() {
   }
 
   @Override
-  public Statistics visitUnknown(LogicalExpression e, Void value) throws RuntimeException {
+  public Statistics<T> visitUnknown(LogicalExpression e, Void value) throws RuntimeException {
     // do nothing for the unknown expression
     return null;
   }
 
   @Override
-  public Statistics visitTypedFieldExpr(TypedFieldExpr typedFieldExpr, Void value) throws RuntimeException {
-    final ColumnStatistics columnStatistics = columnStatMap.get(typedFieldExpr.getPath());
+  public Statistics<T> visitTypedFieldExpr(TypedFieldExpr typedFieldExpr, Void value) throws RuntimeException {
+    final ColumnStatistics<T> columnStatistics = columnStatMap.get(typedFieldExpr.getPath());
     if (columnStatistics != null) {
       return columnStatistics.getStatistics();
     } else if (typedFieldExpr.getMajorType().equals(Types.OPTIONAL_INT)) {
       // field does not exist.
       IntStatistics intStatistics = new IntStatistics();
       intStatistics.setNumNulls(rowCount); // all values are nulls
-      return intStatistics;
+      return (Statistics<T>)intStatistics;
     }
     return null;
   }
 
   @Override
-  public Statistics visitIntConstant(ValueExpressions.IntExpression expr, Void value) throws RuntimeException {
-    return getStatistics(expr.getInt());
+  public Statistics<T> visitIntConstant(ValueExpressions.IntExpression expr, Void value) throws RuntimeException {
+    return (Statistics<T>)getStatistics(expr.getInt());
```

Comment:
It was much nicer when we did not have to explicitly cast everywhere to `(Statistics<T>)`. Please consider re-working this part your changes.

Reply:
I can't completely avoid casting due to Java generics and parquet-mr `Statistics` class limitations. In case changes are OK, let me know as I need to reorder commits.